zilencer/views.py

```diff
@@ -858,25 +858,28 @@ def update_remote_realm_data_for_server(
     RemoteRealmAuditLog.objects.bulk_create(remote_realm_audit_logs)
 
 
-def get_human_user_realm_uuids(realms: List[RealmDataForAnalytics]) -> List[UUID]:
-    billable_realm_uuids = []
-    for realm in realms:
-        # TODO: Remove the `zulipinternal` string_id check once no server is on 8.0-beta.
-        if (
-            realm.is_system_bot_realm
-            or realm.deactivated
-            or realm.host.startswith("zulipinternal.")
-            or (settings.DEVELOPMENT and realm.host.startswith("analytics."))
-        ):  # nocoverage
-            continue
-        billable_realm_uuids.append(realm.uuid)
+def get_human_user_realm_uuids(
+    server: RemoteZulipServer,
+) -> List[UUID]:
+    query = RemoteRealm.objects.filter(
+        server=server,
+        realm_deactivated=False,
+        registration_deactivated=False,
```

Comment:
Added `registration_deactivated` as an extra thing to filter for, that wasn't done in the previous logic, since I think that's more correct.